assets/app/scripts/controllers/catalog/images.js

```diff
@@ -12,9 +12,26 @@ angular.module('openshiftConsole')
     $scope.projectImageRepos = {};
     $scope.openshiftImageRepos = {};
     $scope.builders = [];
-    $scope.images = [];
+    $scope.nonBuilders = [];
     $scope.sourceURL = $routeParams.builderfor;
 
+    var isBuilder = function(imageRepo, tag) {
+      if (!imageRepo.spec || !imageRepo.spec.tags) {
+        return false;
+      }
+
+      var i, imageTags, categoryTags;
+      for (i = 0; i < imageRepo.spec.tags.length; i++) {
+        imageTags = imageRepo.spec.tags[i];
+        if (imageTags.name === tag && imageTags.annotations && imageTags.annotations.tags) {
```

Comment:
Once we find the tag in the spec with a matching name, if it doesn't have a builder annotation, shouldn't we return false, rather than continuing through the list?